server/cache.go

```diff
@@ -272,10 +277,18 @@ func (r *regionsInfo) updateRegion(region *metapb.Region) {
 
 	oldItem := r.searchRegions.ReplaceOrInsert(item)
 	if oldItem == nil {
-		log.Fatalf("updateRegion for none existed region - %v", region)
+		log.Fatalf("updateRegion for none existed region in searchRegions - %v", region)
```

Comment:
Because we check region existence before calling these functions, so if the region doesn't exist here, it should be a bug.